lib/double-fetch-detection.c

```diff
@@ -0,0 +1,34 @@
+#include "linux/double-fetch-detection.h"
+#include <linux/context_tracking.h>
+#include <linux/entry-common.h>
+#include <linux/livepatch.h>
+#include <linux/audit.h>
+#include <linux/sched.h>
+#include <linux/slab.h>
+void addAddress(void* addr, unsigned long leng){
+           if(current->noRead==100 || current->noRead==-1)
```

Comment:
Nit: the name `noRead` has a connotation with something being not read. How about `num_read`?